### PR TITLE
improvements

### PR DESCRIPTION
**Pull request by scoopex**
- cache monitoring data
- cache login token
- improve template

### CODE REVIEW

**Review by tristanlt — COMMENTED** [review]
You propose to change application name "RocketChat" with "Custom - service - RocketChat". Is this a Zabbix convention ? Is there a convention ? it's a very good idea, but if every contributor propose own convention, it's a wild.

I will adopt this convention for our monitoring system. Since don't change anything, it's just a comment.

**Review by tristanlt — DISMISSED** [review]
Storing auth credentials inside pickle is a good idea. However, pickle file was created with read permission for all users. 

Could you set umask before dump ? ( something like that )

```
        try:
            os.umask(0077)
            data["credentials"] = credentials
            pickle.dump(data, open(statusfile,  "wb"))
        except Exception as e:
            logging.critical("but storing status to %s failed : %s" % (statusfile, e))
```

File: server/externalscripts/rocketchat-stats [hunk 1]
```diff
@@ -61,41 +68,87 @@ def login(api_url, username, password):
...
+            return None
+
+        try:
+            data = r.json()
+        except Exception as e:
+            logging.critical("Wrong JSON receive from API /v1/statistics, check endpoint (statistics step): %s" % e)
+            return None
+
+        try:
+            data["credentials"] = credentials
```
Comment (tristanlt) [inline server/externalscripts/rocketchat-stats:137]:
```suggestion
            os.umask(0o077)
            data["credentials"] = credentials
```
Prevent authentication token file read was allowed for all users.

File: RocketChat-Stats.xml [hunk 2]
```diff
@@ -19,7 +19,7 @@
             </groups>
             <applications>
                 <application>
-                    <name>RocketChat</name>
+                    <name>Custom - Service - Rocketchat</name>
```
Comment (tristanlt) [inline RocketChat-Stats.xml:22]: You propose to change application name "RocketChat" with "Custom - service - RocketChat". Is this a Zabbix convention ? Is there a convention ? it's a very good idea, but if every contributor propose own convention, it's a wild.

However, I will adopt this convention for our monitoring system, so don't change anything. At this time, we haven't convention and it's wild ;)

Comment (scoopex) [reply RocketChat-Stats.xml:22]: Just rename it as you prefer - thats our internal convention which we use to distinguish internally modified templates.

File: server/externalscripts/rocketchat-stats (hunk 1 above)
Comment (scoopex) [reply server/externalscripts/rocketchat-stats:137]: fully agree

Comment (tristanlt) [reply server/externalscripts/rocketchat-stats:137]: Great

Comment (tristanlt) [reply server/externalscripts/rocketchat-stats:137]: Dear @scoopex,
Could you integrate this suggested change to PR, it seems I can’t... (1st time using « suggested change »). Many thanks

**Review by tristanlt — APPROVED** [review]
Many thanks

**Pull request merged**